src/app/components/simulation/animation-view.tsx

```diff
@@ -11,153 +11,136 @@ interface IProps {
   isRunning: boolean;
 }
 
-export const AnimationView: React.FC<IProps> = ({light, water, co2Amount, time, isRunning}) => {
-  const co2None = (co2Amount === InputAmount.None);
-  const co2Low = (co2Amount === InputAmount.Some);
-  const co2Normal = (co2Amount === InputAmount.Full);
+export const AnimationView: React.FC<IProps> = ({light, water, co2Amount, time}) => {
+  const noLight = (light === InputAmount.None);
+  const someLight = (light === InputAmount.Some);
+  const fullLight = (light === InputAmount.Full);
+
+  const noWater = (water === InputAmount.None);
+  const someWater = (water === InputAmount.Some);
+  const fullWater = (water === InputAmount.Full);
+
+  const noCo2 = (co2Amount === InputAmount.None);
+  const someCo2 = (co2Amount === InputAmount.Some);
+  const fullCo2 = (co2Amount === InputAmount.Full);
+
   let plantClass = "";
-  let rootClass = "";
+
+  // Cases 7, 8, 9, 16, 17, 18, 25, 26, 27
+  const isNoGrowthCase = noWater;
+  // Cases 1, 4, 10
+  const isCase1OrSame = (noLight && fullWater && fullCo2) || (noLight && someWater && fullCo2) || (noLight && fullWater && someCo2);
+  // Cases 2, 6
+  const isCase2OrSame = (someLight && fullWater && fullCo2) || (fullLight && someWater && fullCo2);
+  // Cases 13, 19, 20, 21, 22, 23, 24
+  const isCase13OrSame = (noLight && someWater && someCo2) || noCo2;
+  // Cases 14, 15
+  const isCase14OrSame = (someWater && someLight && someCo2) || (fullLight && someWater && someCo2);
+
+  const isCase3 = fullLight && fullWater && fullCo2;
+  const isCase5 = someLight && someWater && fullCo2;
+  const isCase11 = someLight && fullWater && someCo2;
+  const isCase12 = fullLight && fullWater && someCo2;
 
   if (time >= 0 && time <= 0.142){ // Day 00
```

Comment:
You don't have to implement this but it might be more readable and self-documenting if you used something like:

```
const numDays = 28;
const timePerDay = 1/numDays;
const timeForDay = (day: number) => day * timePerDay;

if (time < 0) {
  // noop
} else if (time <= timeForDay(4)) {
   ... current day 00 code ...
} else if (time <= timeForDay(8) {
   ... current day 04 code ...
```